src/coreclr/jit/importercalls.cpp

```diff
@@ -1622,6 +1708,138 @@ GenTree* Compiler::impFixupCallStructReturn(GenTreeCall* call, CORINFO_CLASS_HAN
 #endif // FEATURE_MULTIREG_RET
 }
 
+//-----------------------------------------------------------------------------------
+//  impCanSubstituteSig: Checks whether it's safe to replace a call with another one.
+//  This DOES NOT check if the calls are actually compatible, it only checks if their trees are.
+//  Use ONLY when code will call the method with target sig anyway.
+//
+//  Arguments:
+//    sourceSig      - original call signature
+//    targetSig      - new call signature
+//    transformation - transformations performed on the original signature
+//
+//  Return Value:
+//    Whether it's safe to change the IR to call the target method
+//
+bool Compiler::impCanSubstituteSig(CORINFO_SIG_INFO* sourceSig,
+                                   CORINFO_SIG_INFO* targetSig,
+                                   var_types         sourceThis,
+                                   var_types         targetThis)
+{
+    assert(sourceSig->hasImplicitThis() || sourceThis == TYP_UNDEF);
```

Comment:
Seems like `sourceThis` is always `TYP_UNDEF`  and `targetThis` is never `TYP_VOID`?

I can't tell if you are trying to future-proof this method and have it handle cases it currently won't be asked to handle, or if this is a vestige of some earlier version.

Since there is currently just one caller perhaps scope this down to just the cases you'll actually see, and assert if anything else comes along?